src/Corrade/Utility/Sha1.cpp

```diff
@@ -65,20 +65,24 @@ Sha1& Sha1::operator<<(const std::string& data) {
         }
 
         /* Append few last bytes to have the buffer at 64 bytes */
-        _buffer.append(data.substr(0, dataOffset));
+        _buffer.append(data.prefix(dataOffset));
```

Comment:
Nope, this won't work as soon as there is a 0 byte in the data. Same below. (Doesn't the test check this? It should!)

If you want to get fancy, the `_buffer` could be changed to a growable Array .. and there the append takes an arrayview, so this would work again.

Reply:
I guess that's a great opportunity to learn the growable Array API!

Reply:
It is not tested. I adapted `exact64bytes()` to test this case, computing the expected hash with the master version.

Reply:
Actually, this might require a specific use test case, since this will only fail if the `\0` is in the left-over part

Reply:
yep, exactly .. nontrivial to test, but also nontrivial to discover when it gets broken, so a test is critical (thanks in advance!)

Reply:
Okay, so actually, this only happens in iterative use, if the previous data had lefovers and the new data contains a zero in the first `64-numLeftOverBytes` bytes.

Reply:
@mosra So what I've discovered after moving the code to `char _buffer[128]` is that this substr() call up there ads a 0 delimiter to the end, which is not indended, correct?

The other substr call actually works for all the tests, because they end with 0 anyway (input is always a string).

I'll try to figure out how to get the intended hashes with a third-party application

Reply:
To generate the expected hashes, I added Sha1Test.py, it will print the expected hash per test case.